### DISCUSSION TITLE
"No install" challenge

### DISCUSSION THREAD

**Discussion by BooneyNoobington** [azerothcore/azerothcore-wotlk#11231] [Ideas]
Hi all,

when I first tried to install then TrinityCore on Windows, I was surprised how much applications the guide told me to install and I thought "It should be possible to compile TC (or AC for that matter) with much less installation going on."

It turned out that it is quite easy to avoid installing git, cmake and mysql. And boost as well as OpenSSL can be compiled from source with somewhat reasonable effort.

Not that you should do it in that way. I was just wondering if one could.

So I came up with the challenge to compile AC without any installation, defined as the execution of an installer executable which not only copies the necessary files but also performs some sort of configuration - either by creating environment variables or making registry entries. In short: unzipping is fine, clicking on anything .exe or .msi is not.

The most obvious roadblock is the compiler. But Code::Blocks brings with it an implementation of MinGW, so g++ and some sort of make. CMake on the other hand can generate projects for Code::Blocks.

So I downloaded portable versions and compiled boost as well as this OpenSSL project (https://github.com/janbar/openssl-cmake), designed with CMake as the build system.

But when I try to configure the AC sourcecode via CMake, it says: 

```
CMake Error at D:/apps/cmake-3.23.0-rc5-windows-x86_64/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find OpenSSL (missing: OPENSSL_LIBRARIES)
Call Stack (most recent call first):
  D:/apps/cmake-3.23.0-rc5-windows-x86_64/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  src/cmake/macros/FindOpenSSL.cmake:190 (find_package_handle_standard_args)
  deps/openssl/CMakeLists.txt:17 (find_package)
```

So my question is: Is it possible / feasible to compile AC in the described way on Windows or is it just downright impossible?

As I said: I don't plan on actually using AC that way, I'm just wondering.

Cheers
Markus

**Comment by Kitzunu** [comment]
Just want to put it out that any other compiler than MSVC is not supported for windows.

As for your error, CMake simply can't find OpenSSL. Try defining the path in env vars otherwise point to it directly in CMake

**Reply by BooneyNoobington** [reply]
Does not supported also mean impossible? It seeams that [TrinityCore](https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130053/MinGW+GCC+toolchain+Win) could potentially compile with MinGW.

**Reply by Kitzunu** [reply]
Doesn't mean it is impossible, just means that nobody has done it. Just means that none of the organizations maintainers has bothered with anything else than MSVC

**Comment by Nyeriah** [comment]
Honest question, what's the difference between installing them and using a portable version? Isn't the install process basically just about writing system variables to prevent errors like the one you got? Does it do more than just copying files into the install folders + writing register keys?

Furthermore, wouldn't it be easier to install it through docker in that case? With docker you don't need to install any dependency at all ever.

**Reply by BooneyNoobington** [reply]
First of all: I think it's great that AC comes as docker image. Unfortunately I would have to install Docker or WSL since docker can't really be portable as far as I know (otherwise this would be not only the solution to the challenge but also a great way for personal use of AC).

I must admit, I don't really understand what the OpenSSL installer acutally does besides copying files.

**Comment by cbunting99** [comment]
I'm not sure how up to date this is but installing into VM seems pretty straight forward if you are just looking to play around or test the server. I could install this in Linux or compile, but I've been away from programming for many years now. But I'm looking to get AC running myself and this seems pretty easy. https://github.com/nolt/acore-box
